core/store/src/flat/inlining_migration.rs

```diff
@@ -7,16 +10,70 @@ use near_primitives::hash::CryptoHash;
 use near_primitives::shard_layout::ShardUId;
 use tracing::{debug, info};
 
+use crate::flat::store_helper::set_flat_state_values_inlining_migration_status;
 use crate::metrics::flat_state_metrics::inlining_migration::{
     FLAT_STATE_PAUSED_DURATION, INLINED_COUNT, INLINED_TOTAL_VALUES_SIZE, PROCESSED_COUNT,
     PROCESSED_TOTAL_VALUES_SIZE, SKIPPED_COUNT,
 };
 use crate::{DBCol, Store, TrieDBStorage, TrieStorage};
 
-use super::store_helper::decode_flat_state_db_key;
-use super::types::INLINE_DISK_VALUE_THRESHOLD;
+use super::store_helper::{
+    decode_flat_state_db_key, get_flat_state_values_inlining_migration_status,
+};
+use super::types::{FlatStateValuesInliningMigrationStatus, INLINE_DISK_VALUE_THRESHOLD};
 use super::{FlatStateValue, FlatStorageManager};
 
+pub struct FlatStateValuesInliningMigrationHandle {
+    handle: JoinHandle<()>,
+    keep_running: Arc<AtomicBool>,
+}
+
+const BACKGROUND_MIGRATION_READ_STATE_THREADS: usize = 8;
```

Comment:
Maybe reuse `client_background_migration_threads` from config if one wants to speed up migration? Not a strong opinion.